app/models/observation.rb

```diff
@@ -193,6 +239,151 @@ class Observation < AbstractModel
+    case name
+    when String
+      name = Name.find_by(text_name: name)
+    when Integer
+      name = Name.find_by(id: name)
+    end
+    return Observation.none unless name.is_a?(Name)
+
+    # Filter args may add to an array of names to collect Observations
+    names_array = [name]
+    # Maybe add synonyms (Name#synonyms includes original name)
+    names_array = name.synonyms if args[:include_synonyms]
+    # Keep names_array intact as is; maybe add more to its clone name_ids.
+    # (I'm thinking it's easier to pass name ids to the Observation query)
+    name_ids = names_array.map(&:id)
+
+    # Add subtaxa to name_ids array. Subtaxa of synonyms too, if requested
+    # (don't modify the names_array we're iterating over)
+    if args[:include_subtaxa]
+      names_array.each do |n|
+        # |= don't add duplicates
+        name_ids |= Name.subtaxa_of(n).map(&:id)
+      end
+    end
+
+    # Query, with possible join to Naming. Mutually exclusive options:
+    if args[:include_all_name_proposals]
+      joins(:namings).where(namings: { name_id: name_ids })
+    elsif args[:of_look_alikes]
+      joins(:namings).where(namings: { name_id: name_ids }).
+        where.not(name: name_ids)
+    else
+      where(name_id: name_ids)
+    end
+  }
+  scope :of_name_like,
+        ->(name) { where(name: Name.text_name_includes(name)) }
+  scope :by_user, ->(user) { where(user: user) }
+  scope :with_location, -> { where.not(location: nil) }
+  scope :without_location, -> { where(location: nil) }
+  scope :at_location, ->(location) { where(location: location) }
+  scope :in_region,
+        ->(where) { where(Observation[:where].matches("%#{where}")) }
+  # TODO: Reexamine this:
+  #  If Obs lacks lat/lon, should it fall back on Location,
+  #  i.e., Location.in_box(<Obs.location boundaries>)
+  #  And it's buggy (among other things, doesn't cover box straddling 180 def)
+  scope :in_box, # Use named parameters (n, s, e, w), any order
+        lambda { |**args|
+          if args[:s].present? && args[:n].present? &&
+             args[:w].present? && args[:e].present? &&
+             (args[:w] < args[:e])
+
+            where(
+              (Observation[:lat] >= args[:s]).
+              and(Observation[:lat] <= args[:n]).
+              and(Observation[:long] >= args[:w]).
+              and(Observation[:long] <= args[:e])
+            )
+          else
+            Observation.none
+          end
+        }
+  scope :is_collection_location, -> { where(is_collection_location: true) }
+  scope :not_collection_location, -> { where(is_collection_location: false) }
+  scope :with_image, -> { where.not(thumb_image: nil) }
+  scope :without_image, -> { where(thumb_image: nil) }
+  scope :with_notes, -> { where.not(notes: Observation.no_notes) }
+  scope :without_notes, -> { where(notes: Observation.no_notes) }
+  scope :has_notes_field,
+        ->(field) { merge(Observation.notes_include(field.to_sym)) }
```

Comment:
@pellaea Is this OK? Compare to the following in `app/classes/query/modules/conditions.rb`
```ruby
      def notes_field_presence_condition(field)
        field = field.dup
        pat = if field.gsub!(/(["\\])/) { '\\\1' }
                "\":#{field}:\""
              else
                ":#{field}:"
              end
        "observations.notes like \"%#{pat}%\""
      end
```

Reply:
Since we don't have to protect quotes using AR / Arel, we can probably get away with just adding the colons.  I see no advantage to using notes_include, by the way.  I would personally recommend something like this:

`->(field) { where(Observation[:notes].matches("%:#{field}:%")) }`

Reply:
Thanks. That's much better than my still-buggy "fix".